command/agent/job_endpoint.go

```diff
@@ -384,14 +384,18 @@ func (s *HTTPServer) jobUpdate(resp http.ResponseWriter, req *http.Request,
 		return nil, CodedError(400, "Job ID does not match name")
 	}
 
-	// Http region takes precedence over hcl region
+	// Region in api request query param takes precedence over region in job config
 	if args.WriteRequest.Region != "" {
 		args.Job.Region = helper.StringToPtr(args.WriteRequest.Region)
 	}
 
```

Comment:
This is tricky because we would still want to job's region to take precedence if it is set to a non default value. So, the logic should be - _If the writeRequest's region is empty, and the job's region is "global", then use the region of the agent's config and set that as the job's region._ Something like 

```
if args.WriteRequest.Region == ""  && args.Job.Region == "global" {
  args.Job.Region = s.agent.config/Region
}
```